test/regex.jl

```diff
@@ -136,6 +136,24 @@
         @test r"this|that"^2 == r"(?:this|that){2}"
     end
 
+    @testset "iterate" begin
+        m = match(r"(.) test (.+)", "a test 123")
+        @test first(m) == "a"
+        @test collect(m) == ["a", "123"]
+        @test for (i, capture) in m
```

Comment:
I'm confused here: does `m` iterate just the capture values or does it iterate `(index, capture)` pairs?

Reply:
```suggestion
        @test for (i, capture) in enumerate(m)
```

Reply:
Iterates capture values, I just forgot the `enumerate`.
This is what i get for not running tests locally 😂